src/coreclr/nativeaot/Runtime/Full/CMakeLists.txt

```diff
@@ -63,6 +66,23 @@ add_custom_command(
 
 set_source_files_properties("${CMAKE_CURRENT_BINARY_DIR}/AsmOffsets.inc" PROPERTIES GENERATED TRUE)
 
+set_source_files_properties(${SHARED_EVENTPIPE_SOURCES} PROPERTIES LANGUAGE CXX)
+set_source_files_properties(${SHARED_DIAGNOSTIC_SERVER_SOURCES} PROPERTIES LANGUAGE CXX)
+
+if(CLR_CMAKE_HOST_UNIX)
+  if (CMAKE_VERSION VERSION_GREATER 3.11 OR CMAKE_VERSION VERSION_EQUAL 3.11)
+    set_source_files_properties(${SHARED_EVENTPIPE_SOURCES} PROPERTIES COMPILE_OPTIONS -xc++)
+    set_source_files_properties(${SHARED_DIAGNOSTIC_SERVER_SOURCES} PROPERTIES COMPILE_OPTIONS -xc++)
+  endif()
+endif(CLR_CMAKE_HOST_UNIX)
```

Comment:
Do we need to do this? Mono doesn't compile the C files as C++ - the fact that CoreCLR does it is pretty disturbing.

Reply:
This flag is needed to build successfully for CoreCLR and NativeAOT, primarily cause the container code used by the EventPipe common code is C++. Opened #81002 to track investigate removing this flag once the common container code is used.